flocker/provision/_ssh/_keys.py

```diff
@@ -0,0 +1,82 @@
+# Copyright Hybrid Logic Ltd.  See LICENSE file for details.
+# -*- test-case-name: flocker.provision.test.test_ssh_keys -*-
+
+"""
+Tools for dealing with ssh keys.
+"""
+
+import os
+
+from characteristic import attributes, Attribute
+
+from eliot import start_action, Message
+from eliot.twisted import DeferredContext
+
+from twisted.internet.defer import fail
+from twisted.internet.endpoints import UNIXClientEndpoint, connectProtocol
+from twisted.conch.ssh.agent import SSHAgentClient
+from twisted.conch.ssh.keys import Key
+
+
+class AgentNotFound(Exception):
+    """
+    Raised if there is not a running ssh-agent.
+    """
+
+
+@attributes([
+    Attribute("expected_key", instance_of=Key),
+], apply_immutable=True)
+class KeyNotFound(Exception):
+    """
+    Raised if the given key is not in the running ssh-agent.
+    """
+    def __str__(self):
```

Comment:
I don't think it is worthwhile to test this. (The test would essentially be exactly the same as the code it is testing.